portscanner/scanner.go

```diff
@@ -0,0 +1,113 @@
+package portscanner
+
+import (
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	ap "github.com/drypycode/port-scanner/argparse"
+	. "github.com/drypycode/port-scanner/progressbar"
+)
+
+// SafeSlice ...
+type SafeSlice struct {
+	mu 			sync.Mutex
```

Comment:
If you use a https://golang.org/pkg/sync/#RWMutex here, you can perform more granular locking for read/write sections. For example, in length(), you can do ss.mu.RLock(), which won't block writers during a length() call (just reading from OpenPorts)